### DISCUSSION TITLE
OELint: Inconsisten behaviour when including files with variable in name

### DISCUSSION THREAD

**Discussion by pasweistorz** [priv-kweihmann/oelint-adv#627] [Is this a bug, or a feature?]
Hi,

**Describe the bug**
I have the following `test-recipe_1.0.bb` recipe:

```
SUMMARY = "Summary"
DESCRIPTION = "Description"

LICENSE = "CLOSED"

SRC_URI = "file://mytest.sh"

MY_TEST_VARIABLE = "mytest"

require ${MY_TEST_VARIABLE}.inc
require ${MACHINE}.inc

do_install() {
    install -D -m 0755 -t ${D}${bindir} ${WORKDIR}/mytest.sh
}
```

I'm adding the `test-recipe` package to an image and build the image for machine `qemux86-64`. Both files `qemux86-64.inc` and `mytest.inc` exist, are empty and reside next to the `test-recipe_1.0.bb` recipe. The image can be built but I get the following OELint SCA issue:

```
/path/to/test-recipe/test-recipe_1.0.bb:12:error:oelint.file.requirenotfound:'${MACHINE}.inc' was not found
```

In contrast, SCA doesn't complain about the `mytest.inc` file.

My assumption is that the statement `_path = stash.ExpandTerm(_file, item.IncName)` in the implementation of the rule only expands variables from the current file. But the `MACHINE` variable isn't defined in the recipe of course.

Is this desired behaviour or a bug?

**Used layer**
- hash of meta-sca layer: 73e6f7e5a1102cd6fe5b4ae2993f50aa83077bea
- poky hash: 4b07a5316ed4b858863dfdb7cab63859d46d1810
- custom layer

Thanks for investigating.

**Comment by priv-kweihmann** [comment]
`MACHINE` expansion is tricky, as it's not a fixed value.
The `require`  here would fail the build if no matching file can be found - which is in itself an error.

My personal take would be to change it to the following

```
# nooelint: oelint.file.requireinclude
include ${MACHINE}.inc
```

as `include` doesn't have that limitation, meaning it skips silently on missing files
